microk8s-resources/actions/common/utils.sh

```diff
@@ -143,13 +143,36 @@ get_ips() {
 
 
 produce_server_cert() {
-    # Produce the server certificate adding the IP passed as a parameter
-    # Parameters:
-    # $1 IP we want in the certificate
+    # Produce the server certificates based on the rendered csr.conf.rendered.
+    # The file csr.conf.rendered is compared with csr.conf to determine if a regeneration of server certs must be done.
+    #
+    # Returns 
+    #  0 if no change
+    #  1 otherwise. 
+
+    render_csr_conf
+    if ! [ -f "${SNAP_DATA}/certs/csr.conf" ];
+    then
+        echo "changeme" >  "${SNAP_DATA}/certs/csr.conf" 
+    fi
+
+    if ! cmp -s "${SNAP_DATA}/certs/csr.conf.rendered" "${SNAP_DATA}/certs/csr.conf"; then
```

Comment:
We would better use cmp with the full path `${SNAP}/usr/bin/cmp`  so we know we always use the one `cmp` we ship.

Reply:
We do not do this for openssl because openssl is a dependency to snapd and it is on the host for sure.

Reply:
Oh i see. I'll get to that.